### DISCUSSION TITLE
Approach to refresh button/action

### DISCUSSION THREAD

**Discussion by jbalatero** [facebookexperimental/Recoil#1678] [Q&A]
Hi guys, I wanted to know if there is a way to tell if all the loadables are in "hasValue".

I have a bunch of selectors dependent on a `forceReload` selector like this:

```
selector({
  key: `${key}/default`,
  get: async ({ get }) => {
x    // force update cached data
    get(forceReload);

    return "hello";
  },
})
```

and I wanted to start a global loading element and stops upon completion(hasValue) of all the dependent selectors.

**Comment by drarmstr** [comment]
Check out the `waitForAll()` and related helpers.
